cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -629,13 +674,25 @@ func (m *stateMachine) setUnlockStateWithFallbackKey(partName string, unlockRes
 		} else {
 			// if we don't have an encrypted device and err != nil, then the
 			// device must be not-found, see above checks
-			m.degradedState.LogErrorf("cannot locate %s partition: %v", partName, err)
+			// it's valid to not need ubuntu-save with unencrypted partitions,
+			// so we can not report an error for ubuntu-save not being found if
+			// not encrypted, but we must then also set it as absent-but-optional
+			if partName != "ubuntu-save" {
```

Comment:
this looks a bit messy, can we make this more a problem of the caller instead ? seems we need a flag like we have on setFindState ?

Reply:
yes I think we can do this better, given your other comment I forgot that now PartDevice == "" means that we for sure didn't find it

Reply:
hopefully this is more clear now